### PR TITLE
Ravinder | Stage | New Inventory Code

### PR DESCRIPTION
**Pull request by ravinderwalkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/inventory/components/sidebar-components/stockList.component.ts [hunk 1]
```diff
@@ -23,9 +23,9 @@ import { BreakpointObserver } from '@angular/cdk/layout';
             <span class="d-block mr-r1" *ngIf="item.count" [hidden]="(activeStockUniqueName$ | async) === item.uniqueName">
          {{item.count}}</span>
           </a>
-          <button class="btn btn-link btn-xs pull-right" (click)="goToManageStock(item)" *ngIf="!isMobileScreen && (activeStockUniqueName$ | async) === item.uniqueName">
+          <a class="btn btn-link btn-xs pull-right" [routerLink]="'/pages/new-inventory/stock/edit/' + item.uniqueName" *ngIf="!isMobileScreen && (activeStockUniqueName$ | async) === item.uniqueName">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/inventory/components/sidebar-components/stockList.component.ts]: Please revert the code according to old inventory work

File: apps/web-giddh/src/app/inventory/components/stock-report-component/inventory.stockreport.component.html [hunk 2]
```diff
@@ -36,7 +36,12 @@ <h2 class="head-report" *ngIf="stockReport && stockReport.stockUnit">
 
             <div class="right-inventory">
                 <div class="btn-group" dropdown [autoClose]="true" placement="bottom right">
-                    <button class="btn-link icon-btn mr-r15" (click)="editStock()" tooltip="Edit Stock">
+                    <button
+                        class="btn-link icon-btn mr-r15"
+                        href="javascript:;"
+                        [routerLink]="'/pages/new-inventory/stock/edit/' + stockUniqueName"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/inventory/components/stock-report-component/inventory.stockreport.component.html]: Please revert the code according to old inventory work

File: apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    @ViewChild('datepickerTemplate') public datepickerTemplate: ElementRef;
+
+    constructor(
+        private generalService: GeneralService, private modalService: BsModalService,
+        private breakPointObservar: BreakpointObserver) {
+    }
+
+    /*datepicker funcation*/
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.ts:51]: Improper JS doc

File: apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        /* added image path */
+        this.imgPath = (isElectron || isCordova) ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
+
+        this.breakPointObservar.observe([
+            '(max-width: 767px)'
+        ]).pipe(takeUntil(this.destroyed$)).subscribe(result => {
+            this.isMobileScreen = result.matches;
+        });
+    }
+    /* It will show/hide hsn code field */
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.ts:104]: Improper JS doc

File: apps/web-giddh/src/app/new-inventory/component/about-combo-detail/about-combo-detail.component.scss [hunk 5]
```diff
@@ -0,0 +1,74 @@
+@import "../../../../assets/css/color.scss";
```
Comment (shivanshRawat14) [inline apps/web-giddh/src/app/new-inventory/component/about-combo-detail/about-combo-detail.component.scss]: Remove this line

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    /* This will store the x/y position of the field to show datepicker under it */
+    public dateFieldPosition: any = { x: 0, y: 0 };
+    @ViewChild('datepickerTemplate') public datepickerTemplate: ElementRef;
+
+    constructor(
+        private generalService: GeneralService,
+        private modalService: BsModalService
+    ) { }
+
+    /*datepicker funcation*/
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts:41]: Improper JS doc

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        this.hideGiddhDatepicker();
+        if (value && value.startDate && value.endDate) {
+            this.selectedDateRange = { startDate: moment(value.startDate), endDate: moment(value.endDate) };
+            this.selectedDateRangeUi = moment(value.startDate).format(GIDDH_NEW_DATE_FORMAT_UI) + " - " + moment(value.endDate).format(GIDDH_NEW_DATE_FORMAT_UI);
+            this.fromDate = moment(value.startDate).format(GIDDH_DATE_FORMAT);
+            this.toDate = moment(value.endDate).format(GIDDH_DATE_FORMAT);
+        }
+    }
+    /* Aside pane toggle fixed class */
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts:84]: Improper JS doc

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    public toggleBodyClass(): void {
+        if (this.asideMenuState === 'in') {
+            document.querySelector('body').classList.add('fixed');
+        } else {
+            document.querySelector('body').classList.remove('fixed');
+        }
+    }
+    /* Create combo aside pane open function */
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts:92]: Improper JS doc

File: apps/web-giddh/src/app/new-inventory/component/combo-list/inventory-combo-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+})
+
+export class InventoryComboListComponent implements OnInit {
+    /* this will store image path*/
+    public imgPath: string = '';
+
+    public ngOnInit() {
+        /* added image path */
+        this.imgPath = (isElectron || isCordova) ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/combo-list/inventory-combo-list.component.ts]: isCordova is not needed

File: apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.html [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+                            *dropdownMenu
+                            class="dropdown-menu"
+                            role="menu"
+                            aria-labelledby="more-btn-inventory"
+                        >
+                            <li role="menuitem">
+                                <a
+                                    class="dropdown-item"
+                                    href="javascript:void(0);"
+                                    [routerLink]="['/pages/new-inventory/adjust-inventory']"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.html:40]: Please check this

File: apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.html [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+                                <a class="dropdown-item text-primary" *ngIf="isMobileScreen" href="javascript:void(0);"
+                                    >+ Add Inventory</a
+                                >
+                            </li>
+                        </ul>
+                    </div>
+                    <button
+                        class="btn btn-sky-blue bdr-radius-50"
+                        *ngIf="!isMobileScreen"
+                        [routerLink]="['/pages/new-inventory/create-new-inventory']"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/about-group-detail/about-group-detail.component.html:57]: Please check this

File: apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.html [hunk 12]
```diff
@@ -0,0 +1,4977 @@
+<div class="relative create-new-inventory-page static-tabs-on-page">
+    <hamburger-menu [pageHeading]="'Create Inventory'"></hamburger-menu>
+    <div class="new-inventory-page-header">
+        <label class="page-title-inventory"> Create Inventory </label>
+
+        <div class="inventory-header-right">
+            <button mat-icon-button class="ml-3" style="font-size: 25px" [routerLink]="['/pages/new-inventory']">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.html:7]: Please check this

File: apps/web-giddh/src/app/new-inventory/component/inventory-dashboard/inventory-dashboard.component.ts [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    constructor() {
+    }
+
+    /* This will hold the value out/in to open/close setting sidebar popup */
+    public asideInventorySidebarMenuState: string = 'in';
+
+    /* Aside pane state*/
+    public asideMenuState: string = 'out';
+
+    /**
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/inventory-dashboard/inventory-dashboard.component.ts:19]: Invalid JS doc

File: apps/web-giddh/src/app/services/apiurls/inventory.api.ts [hunk 14]
```diff
@@ -63,5 +64,15 @@ export const INVENTORY_API = {
     DELETE_BRANCH_TRANSFER: 'company/:companyUniqueName/branch-transfer/:branchTransferUniqueName',
     UPDATE_BRANCH_TRANSFER: 'company/:companyUniqueName/branch-transfer/:branchTransferUniqueName',
     DOWNLOAD_NEW_BRANCH_TRANSFER: 'company/:companyUniqueName/branch-transfer/download?fileType=base64',
-    GET_UNIT_CODE_REGEX: 'ui/forms?formName=:formName&country=:country'
+    GET_UNIT_CODE_REGEX: 'ui/forms?formName=:formName&country=:country',
+    V5: {
+        CREATE_STOCK_GROUP: COMMON_V5 + '/stock-group',
+        GET_STOCK_GROUP: COMMON_V5 + '/stock-group/:groupUniqueName',
+        UPDATE_STOCK_GROUP: COMMON_V5 + '/stock-group/:groupUniqueName'
+    },
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/apiurls/inventory.api.ts:72]: Please check this

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 15]
```diff
@@ -896,4 +896,107 @@ export class InventoryService {
...
+     *
+     * @param {*} model
+     * @returns {Observable<BaseResponse<any, any>>}
+     * @memberof InventoryService
+     */
+    public createStockGroup(model: any): Observable<BaseResponse<any, any>> {
+        const companyUniqueName = this.generalService.companyUniqueName;
+        let url = this.config.apiUrl + INVENTORY_API.V5.CREATE_STOCK_GROUP;
+        url = url.replace(":companyUniqueName", companyUniqueName);
+
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/inventory.service.ts:911]: Please check this file

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 16]
```diff
@@ -0,0 +1,359 @@
...
+import { GeneralService } from "../../../services/general.service";
+import { ToasterService } from "../../../services/toaster.service";
+import { MatAutocompleteTrigger } from "@angular/material/autocomplete";
+@Component({
+	selector: 'stock-balance',
+	templateUrl: './stock-balance.component.html',
+	styleUrls: ['./stock-balance.component.scss'],
+})
+
+export class StockBalanceComponent implements OnInit, OnDestroy {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts:23]: Format this file

File: apps/web-giddh/src/assets/locale/inventory/group-stockreport/en.json [hunk 17]
```diff
@@ -1,16 +0,0 @@
-{
-    "upcoming_feature": "Upcoming feature",
-    "select_branch": "Select Branch",
-    "new_transfer": "New Transfer",
-    "product_name": "Product Name",
-    "search_product": "Search Product",
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/assets/locale/inventory/group-stockreport/en.json:6]: Please check this file

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 18]
```diff
@@ -10261,3 +10583,345 @@ input[type="checkbox"].cdk-visually-hidden::after {
...
+            color: var(--color-blue);
+        }
+    }
+
+    .mat-pseudo-checkbox {
+        color: var(--color-primary-checkbox-and-expansion-panel-arrow);
+    }
+}
+
+.myPanelClass {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/assets/css/style-2.scss:10715]: Avoid camelCase names in class

**Review by the-prateek-mishra — APPROVED** [review]

**Pull request merged**